lib/schema-controller.js

```diff
@@ -1,8 +1,24 @@
 'use strict'
 
 const { buildSchemas } = require('./schemas')
-const SerializerSelector = require('@fastify/fast-json-stringify-compiler')
-const ValidatorSelector = require('@fastify/ajv-compiler')
+
+/**
+ * Inline require to avoid hard dependency on ajv when a custom validator is used
+ */
+function buildDefaultValidator () {
+  const ValidatorSelector = require('@fastify/ajv-compiler')
```

Comment:
We should add a test for it. It must be a new test file and then we can inspect the `require.cache`: https://nodejs.org/api/modules.html#requirecache